fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -309,9 +310,14 @@ public List<PartitionKey> getPrunedPartitions(Table table, ScalarOperator predic
 
             StructLike partitionData = fileScanTask.file().partition();
             List<String> values = PartitionUtil.getIcebergPartitionValues(spec, partitionData);
+            if (values.size() != partitionColumns.size()) {
+                // ban partition evolution and non-identify column.
+                continue;
+            }
             try {
-                partitionKeys.add(createPartitionKey(values, icebergTable.getPartitionColumns(), table.getType()));
+                partitionKeys.add(createPartitionKey(values, partitionColumns, table.getType()));
```

Comment:
for transformed partition column, partitionColumn get the base column, and value get data from partition data(transformed), is that ok?

Reply:
At present, it seems that this is equivalent. maybe when we really use partitionkey, we need to make adjustments here.